### PR TITLE
Hyperbahn: support lazy handling through the service proxy

### PR DESCRIPTION
**Pull request by jcorbin**
Currently will have no effect since no sub channel handler implements `.handleLazily`, this merely puts the delegation in place so that they can Real Soon Now ™.

r @Raynos

### CODE REVIEW

File: node/hyperbahn/service_proxy.js [hunk 1]
```diff
@@ -89,6 +89,58 @@ util.inherits(ServiceDispatchHandler, EventEmitter);
...
+        // TODO: reqFrame.extendLogInfo would be nice, especially if it added
+        // things like callerName and arg1
+        self.channel.logger.warn('missing service name in lazy frame', conn.extendLogInfo({}));
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'missing serviceName');
+        return false;
+    }
+
+    var chan = self.channel.subChannels[serviceName];
+    if (!chan) {
+        chan = self.getOrCreateServiceChannel(serviceName);
```
Comment (Raynos) [inline node/hyperbahn/service_proxy.js]: This is a change in semantics. We never call this method in `handlRequest`.

Consider returning `false` here

File: node/hyperbahn/service_proxy.js [hunk 2]
```diff
@@ -89,6 +89,58 @@ util.inherits(ServiceDispatchHandler, EventEmitter);
...
+
+    var serviceName = res.value;
+    if (!serviceName) {
+        // TODO: reqFrame.extendLogInfo would be nice, especially if it added
+        // things like callerName and arg1
+        self.channel.logger.warn('missing service name in lazy frame', conn.extendLogInfo({}));
+        self._sendLazyErrorFrame(conn, reqFrame, 'BadRequest', 'missing serviceName');
+        return false;
+    }
+
```
Comment (Raynos) [inline node/hyperbahn/service_proxy.js]: // TODO implement isBlocked
// TODO implement rateLimit

Comment (jcorbin) [inline node/hyperbahn/service_proxy.js]: those are TODOs for the next RelayHandle diff, no?  What is `isBlocked`?

Comment (jcorbin) [inline node/hyperbahn/service_proxy.js]: nvm, I found it by RTFC... I see... will adjust

File: node/hyperbahn/service_proxy.js (hunk 1 above)
Comment (jcorbin) [inline node/hyperbahn/service_proxy.js]: nope, this is exactly what handleDefault does, called by handleRequest

**Comment by Raynos** [comment]
do it.

File: node/hyperbahn/service_proxy.js [hunk 3]
```diff
@@ -89,6 +89,46 @@ util.inherits(ServiceDispatchHandler, EventEmitter);
...
 ServiceDispatchHandler.prototype.type = 'tchannel.hyperbahn.service-dispatch-handler';
 
+ServiceDispatchHandler.prototype.handleLazily =
+function handleLazily(conn, reqFrame) {
+    var self = this;
+
+    var res = reqFrame.bodyRW.lazy.readService(reqFrame);
+    if (res.err) {
+        // TODO: stat?
+        self.channel.logger.warn('failed to lazy read frame serviceName', conn.extendLogInfo({
```
Comment (kriskowal) [inline node/hyperbahn/service_proxy.js]: serviceProxy.extendLogInfo is a thing

**Pull request merged**